### PR TITLE
[MO] - [system test] -> @Parallel deletion of Roles, CRBs, CRDs...

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Enhancement / new feature

### Description

This PR accelerate the deletion of our resources, which are created during test cases.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java [hunk 1]
```diff
@@ -271,8 +272,24 @@ public void deleteResources(ExtensionContext testContext) throws Exception {
         if (!STORED_RESOURCES.containsKey(testContext.getDisplayName()) || STORED_RESOURCES.get(testContext.getDisplayName()).isEmpty()) {
             LOGGER.info("In context {} is everything deleted.", testContext.getDisplayName());
         }
-        while (STORED_RESOURCES.containsKey(testContext.getDisplayName()) && !STORED_RESOURCES.get(testContext.getDisplayName()).isEmpty()) {
-            STORED_RESOURCES.get(testContext.getDisplayName()).pop().getThrowableRunner().run();
+
+
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java:276]: Do we need the two empty lines?

**Review by scholzj — APPROVED** [review]
LGTM assuming it works.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java [hunk 2]
```diff
@@ -271,8 +272,24 @@ public void deleteResources(ExtensionContext testContext) throws Exception {
...
+            new AtomicInteger(STORED_RESOURCES.get(testContext.getDisplayName()).size()) :
+            // stack has no elements
+            new AtomicInteger(0);
+        while (STORED_RESOURCES.containsKey(testContext.getDisplayName()) && numberOfResources.get() > 0) {
+            STORED_RESOURCES.get(testContext.getDisplayName()).parallelStream().parallel().forEach(
+                resourceItem -> {
+                    try {
+                        resourceItem.getThrowableRunner().run();
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java:288]: Wouldn't be better to log something into the console? Maybe debug level could be enough

**Review by Frawless — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java:288]: This automatically invokes the delete() method:
```
2021-11-24 11:56:32 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of CustomResourceDefinition kafkamirrormaker2s.kafka.strimzi.io in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of CustomResourceDefinition kafkarebalances.kafka.strimzi.io in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ConfigMap strimzi-cluster-operator in namespace infra-namespace
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-broker-delegation in namespace (not set)
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-client-delegation in namespace (not set)
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-broker-delegation in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of RoleBinding strimzi-cluster-operator in namespace infra-namespace
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-client-delegation in namespace (not set)
```

**Pull request merged**